internal/shims/shims.go

```diff
@@ -368,18 +373,25 @@ func ToolExecutables(conf config.Config, plugin plugins.Plugin, version toolvers
 
 // ExecutablePaths returns a slice of absolute directory paths that tool
 // executables are contained in.
-func ExecutablePaths(conf config.Config, plugin plugins.Plugin, version toolversions.Version) ([]string, error) {
+func ExecutablePaths(conf config.Config, plugin plugins.Plugin, version toolversions.Version) (paths []string, err error) {
 	dirs, err := ExecutableDirs(plugin)
 	if err != nil {
 		return []string{}, err
 	}
 
+	shimsDir, err := os.Stat(path.Join(plugin.Dir, "shims"))
+	if err == nil && shimsDir.IsDir() {
+		paths = append(paths, path.Join(plugin.Dir, "shims"))
+	}
```

Comment:
Are these 4 lines necessary?

Reply:
I've added this code back into https://github.com/asdf-vm/asdf/pull/2076 which addresses the linter warnings present here. Closing this PR in favor of that one (it still contains all your commits).